concurrency-limits-core/src/main/java/com/netflix/concurrency/limits/limiter/AbstractPartitionedLimiter.java

```diff
@@ -134,25 +134,30 @@ void updateLimit(int totalLimit) {
         }
 
         boolean isLimitExceeded() {
-            return busy >= limit;
+            return busy.get() >= limit;
         }
 
-        void acquire() {
-            busy++;
-            inflightDistribution.addSample(busy);
+        int acquire() {
```

Comment:
this is technically a binary incompatible breaking change. Can we instead introduce this as a new `private final int acquireAndGet()` method and `private final void acquirePostSample(int nowBusy)`, while `@Deprecated` the existing `void acquire()` which calls into those methods?

It looks like the big change here is the move to using atomic and splitting up the behavior in the public `acquire` method to more narrowly scope the lock. It's possible I'm missing something to where the new methods couldn't be privately scoped or if the default scoped `acquire` method should change its behavior, but it's not obvious from the change alone.

Reply:
This is a great solution, thanks!

Reply:
fixed in https://github.com/Netflix/concurrency-limits/pull/204/commits/794dac3d35ccf6f1ca586e509c0f3f9164750b75